src/FileSystem/SourceFileFilter.php

```diff
@@ -54,12 +55,22 @@ class SourceFileFilter
      */
     private $filters;
 
-    public function __construct(string $filter)
+    /**
+     * @var string[]
+     */
+    private $excludeDirectories;
+
+    /**
+     * @param string[] $excludeDirectories
+     */
+    public function __construct(string $filter, array $excludeDirectories)
```

Comment:
Note to myself: check the usage of filters, maybe we do that trim + explode operation more than once